redfish-core/lib/license_service.hpp

```diff
@@ -0,0 +1,678 @@
+{
+    res.result(boost::beast::http::status::internal_server_error);
+    addMessageToErrorJson(res.jsonValue, installFailed(reason));
+}
+
+inline nlohmann::json notApplicableToTarget()
+{
+    return getLicenseLog(
+        redfish::registries::license::Index::notApplicableToTarget, {});
+}
+
+static void notApplicableToTarget(crow::Response& res)
+{
+    res.result(boost::beast::http::status::bad_request);
+    addMessageToErrorJson(res.jsonValue, notApplicableToTarget());
+}
+
+} // namespace messages
+
+static std::unique_ptr<sdbusplus::bus::match::match>&
+    getLicenseActivationStatusMatch()
+{
+    static std::unique_ptr<sdbusplus::bus::match::match>
+        licenseActivationStatusMatch;
+    return licenseActivationStatusMatch;
+}
+
+inline void getLicenseEntryCollection(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp)
+{
+    sdbusplus::message::object_path path("/xyz/openbmc_project/license");
+    dbus::utility::getManagedObjects(
+        "xyz.openbmc_project.PLDM", path,
+        [asyncResp](const boost::system::error_code& ec,
+                    const dbus::utility::ManagedObjectType& resp) {
+        if (ec)
+        {
+            BMCWEB_LOG_ERROR("LicenseEntry resp_handler got error {}", ec);
+            // continue here to show zero members
+            asyncResp->res.jsonValue["Members"] = nlohmann::json::array();
+            asyncResp->res.jsonValue["Members@odata.count"] = 0;
+            return;
+        }
+
+        nlohmann::json& entriesArray = asyncResp->res.jsonValue["Members"];
+        entriesArray = nlohmann::json::array();
+
+        for (const auto& object : resp)
+        {
+            std::string entryID = object.first.filename();
+            if (entryID.empty())
+            {
+                continue;
+            }
+            entriesArray.push_back({});
+            nlohmann::json& thisEntry = entriesArray.back();
+            thisEntry["@odata.id"] = "/redfish/v1/LicenseService/Licenses/" +
+                                     entryID;
+            thisEntry["Id"] = entryID;
+            thisEntry["Name"] = entryID + " License Entry";
+            asyncResp->res.jsonValue["Members@odata.count"] =
+                entriesArray.size();
+        }
+    });
+}
+
+inline void requestRoutesLicenseService(App& app)
+{
+    BMCWEB_ROUTE(app, "/redfish/v1/LicenseService/")
+        .privileges({{"Login"}})
+        .methods(boost::beast::http::verb::get)(
+            [](const crow::Request&,
+               const std::shared_ptr<bmcweb::AsyncResp>& asyncResp) {
+        asyncResp->res.jsonValue["@odata.id"] = "/redfish/v1/LicenseService/";
+        asyncResp->res.jsonValue["@odata.type"] =
+            "#LicenseService.v1_0_0.LicenseService";
+        asyncResp->res.jsonValue["Name"] = "License Service";
+        asyncResp->res.jsonValue["Id"] = "LicenseService";
+
+        asyncResp->res.jsonValue["Licenses"] = {
+            {"@odata.id", "/redfish/v1/LicenseService/Licenses"}};
```

Comment:
According to https://github.com/openbmc/bmcweb/blob/master/COMMON_ERRORS.md#13-complete-replacement-of-the-response-object,

```
        asyncResp->res.jsonValue["Licenses"]["@odata.id"] = 
                    "/redfish/v1/LicenseService/Licenses";
```

Reply:
I would prefer not to do this change for rebase PR

Reply:
done